download_clients/pivnet_client.go

```diff
@@ -170,25 +170,47 @@ func (p *pivnetClient) checkForSingleProductFile(glob string, productFiles []piv
 	return nil
 }
 
+func (p *pivnetClient) checkForSingleStemcellType(stemcellSlugs []string) error {
+    if len(stemcellSlugs) > 1 {
+        return fmt.Errorf("multiple stemcell types found: %s", strings.Join(stemcellSlugs, ", "))
+    }
+    return nil
+}
+
 func (p *pivnetClient) getLatestStemcell(dependencies []pivnet.ReleaseDependency) (string, string, error) {
 	var (
-		stemcellSlug string
+		stemcellSlugs []string
 		versions     []string
 	)
 
 	for _, dependency := range dependencies {
 		if strings.Contains(dependency.Release.Product.Slug, "stemcells") {
-			stemcellSlug = dependency.Release.Product.Slug
+			if !contains(stemcellSlugs, dependency.Release.Product.Slug) {
+				stemcellSlugs = append(stemcellSlugs, dependency.Release.Product.Slug)
+			}
 			versions = append(versions, dependency.Release.Version)
 		}
 	}
 
+	if err := p.checkForSingleStemcellType(stemcellSlugs); err != nil {
+        return "", "", fmt.Errorf("could not determine stemcell: %s", err)
+    }
+
 	stemcellVersion, err := getLatestStemcellVersion(versions)
 	if err != nil {
 		return "", "", err
 	}
 
-	return stemcellSlug, stemcellVersion, nil
+	return stemcellSlugs[0], stemcellVersion, nil
+}
+
+func contains(slice []string, str string) bool {
```

Comment:
Didn't pull the branch down, but I think you can use the `slices` package to do contains operations instead of the helper function.

https://freshman.tech/snippets/go/check-if-slice-contains-element/

Reply:
Good call. I've added the newness